integration-tests/chopsticks/src/tests/xcm/limitedReserveTransferSpiritnetHydraDx.test.ts

```diff
@@ -0,0 +1,120 @@
+	const { checkEvents, checkSystemEvents } = withExpect(expect)
+
+	// set storage
+	await spiritnetContext.dev.setStorage(SpiritnetConfig.defaultStorage(keysAlice.address))
+	await hydradxContext.dev.setStorage(HydraDxConfig.defaultStorage(keysAlice.address))
+
+	// Create some new blocks to have consistent snapshots
+	await new Promise((r) => setTimeout(r, 50))
+	await spiritnetContext.dev.newBlock()
+	await hydradxContext.dev.newBlock()
+
+	// check initial balance
+	const balanceSovereignAccountHydraDxBeforeTx = await getFreeBalanceSpiritnet(
+		SpiritnetConfig.hydraDxSovereignAccount
+	)
+	expect(balanceSovereignAccountHydraDxBeforeTx).eq(BigInt(0))
+
+	const omniPoolAddress = u8aToHex(decodeAddress(HydraDxConfig.omnipoolAccount))
+	const signedTx = spiritnetContext.api.tx.polkadotXcm
+		.limitedReserveTransferAssets(
+			SpiritnetConfig.V3.hydraDxDestination,
+			SpiritnetConfig.V3.hydraDxBeneficiary(omniPoolAddress),
+			SpiritnetConfig.V3.nativeAssetIdLocation(KILT),
+			0,
+			'Unlimited'
+		)
+		.signAsync(keysAlice)
+
+	const events = await sendTransaction(signedTx)
+
+	// Produce new blocks
+	// fixes api runtime disconnect warning
+	await new Promise((r) => setTimeout(r, 50))
+	await spiritnetContext.chain.newBlock()
+	await hydradxContext.dev.newBlock()
+
+	// Check events
+
+	checkEvents(events, 'xcmpQueue').toMatchSnapshot('sender events xcm queue pallet')
+	checkEvents(events, 'polkadotXcm').toMatchSnapshot('sender events xcm pallet')
+	checkEvents(events, { section: 'balances', method: 'Withdraw' }).toMatchSnapshot('sender events Balances')
+
+	checkSystemEvents(hydradxContext, { section: 'currencies', method: 'Deposited' }).toMatchSnapshot(
+		'receiver events currencies'
+	)
+	checkSystemEvents(hydradxContext, 'xcmpQueue').toMatchSnapshot('receiver events xcmpQueue')
+
+	// check balance
+	const balanceSovereignAccountHydraDxAfterTx = await getFreeBalanceSpiritnet(SpiritnetConfig.hydraDxSovereignAccount)
+	expect(balanceSovereignAccountHydraDxAfterTx).eq(BigInt(KILT))
+	const freeBalanceOmnipoolAccount = await getFreeBalanceHydraDxKilt(HydraDxConfig.omnipoolAccount)
+	expect(freeBalanceOmnipoolAccount).eq(BigInt(KILT))
+}, 20_000)
+
+test('Limited Reserve V2 Transfers from Spiritnet Account Bob -> HydraDx', async ({ expect }) => {
+	const { checkEvents, checkSystemEvents } = withExpect(expect)
+
+	// Set storage
+	await spiritnetContext.dev.setStorage(SpiritnetConfig.defaultStorage(keysAlice.address))
+	await hydradxContext.dev.setStorage(HydraDxConfig.defaultStorage(keysAlice.address))
+
+	// Create some new blocks to have consistent snapshots
+	await new Promise((r) => setTimeout(r, 50))
+	await spiritnetContext.dev.newBlock()
+	await hydradxContext.dev.newBlock()
+
+	// pre submit extrinsic checks
+	const balanceSovereignAccountHydraDxBeforeTx = await getFreeBalanceSpiritnet(
+		SpiritnetConfig.hydraDxSovereignAccount
+	)
+	expect(balanceSovereignAccountHydraDxBeforeTx).eq(BigInt(0))
```

Comment:
Same here.

Reply:
This is also checked. Please have a look on these lines: https://github.com/KILTprotocol/kilt-node/pull/614/files#diff-ce4305e382897732899a9335dad6b6d4350a23b2d2be29fc5c4ca6f864806a48R117-R118

Reply:
[ea2fa38](https://github.com/KILTprotocol/kilt-node/pull/614/commits/ea2fa38e25cb44e82b4a6ff5607b6b34032302b7)